go-controller/pkg/node/node.go

```diff
@@ -279,12 +280,23 @@ func (n *OvnNode) Start(wg *sync.WaitGroup) error {
 	n.WatchEndpoints()
 
 	// start the cni server
-	cniServer := cni.NewCNIServer("", kclient.KClient)
-	err = cniServer.Start(cni.HandleCNIRequest)
+	n.cniServer = cni.NewCNIServer("", kclient.KClient)
+	err = n.cniServer.Start(cni.HandleCNIRequest)
+	// Watch pods so we can cancel CNI requests when pods are deleted
+	n.WatchPods()
```

Comment:
I'm guessing we already filter the pods just for this node right?

Reply:
yeah, that informer has  a selector for the node

Reply:
I don't think so. WatchPods() is watching all pods afaict. This is the same watcher as the one we run on master afaik.

Reply:
https://github.com/ovn-org/ovn-kubernetes/blob/23bda5e9fdda405ef9aa33a3cb0cfebecf8155a0/go-controller/pkg/factory/factory.go#L219-L229

Reply:
I think that casey has changed it recently,

Reply:
Yeah, Casey made it only watch pods on the node.